### PR TITLE
fix(stacks): Make seeded marimo notebooks recognisable as notebooks

### PR DESCRIPTION
**Pull request by stefanko-ch**
## The symptom

Five of six seeded marimo notebooks showed a plain **code icon** in marimo's file browser instead of a notebook icon, and did not appear in the Workspace tree at all:

```
Workspace
└── Bsc_EDS_..._stefan_koch
    └── nexus_seeds
        └── marimo
            └── Getting_Started_PySpark.py        ← the only one listed
```

Nothing ever failed. The notebooks open and run perfectly by direct URL. The only signal was the icon — which is why this was reported twice as *"I don't see the seeds"* before anyone looked at one.

## The cause

marimo's own rule, from the version this project runs:

```python
# marimo/_server/files/directory_scanner.py
READ_LIMIT = 512
# "Python (.py) files are marimo apps if the header (first 512 bytes)
#  contains both `marimo.App` and `import marimo`."
```

Every seed opens with a long module docstring, which pushes both markers past that window.

| File | `marimo.App` @ byte | before | after |
|---|---|---|---|
| `Getting_Started_PySpark.py` | 419 → 419 | ✅ | ✅ |
| `Getting_Started_Unity_Catalog.py` | 697 → 326 | ❌ | ✅ |
| `Verify_Spark_And_S3.py` | 729 → 325 | ❌ | ✅ |
| `Getting_Started_PostgREST.py` | 1551 → 311 | ❌ | ✅ |
| `Getting_Started_DuckDB.py` | 1619 → 308 | ❌ | ✅ |
| `NYC_Taxi_Pipeline.py` | 2079 → 328 | ❌ | ✅ |

PySpark passed at 419 by luck rather than design — 93 bytes of headroom, and nobody had chosen that.

**Three of the five have been wrong since 2026-08-26 (61a8b6e).** Two are from this week.

## The fix

Keep a short summary docstring — it is what a reader sees on GitHub — and move the prose into each notebook's **first `mo.md` cell**, where a reader of the *notebook* sees it instead. Nothing is dropped; per-file line counts stay level.

`Getting_Started_PySpark.py` is left untouched: it already passes, and the new test is now what catches a future docstring growing past the limit.

Not used: marimo's `# /// script` escape hatch, which makes it read the whole file. That block exists for PEP 723 inline metadata and changes `uv run` behaviour — defeating a limit with it would be a side effect, not a fix.

## Verification

With marimo's **own** function, run inside the deployed marimo container rather than against a reimplementation of the rule:

```
$ docker exec marimo python -c 'from marimo._server.files.directory_scanner import is_marimo_app; ...'
  Getting_Started_DuckDB.py            is_marimo_app = True
  Getting_Started_PostgREST.py         is_marimo_app = True
  Getting_Started_PySpark.py           is_marimo_app = True
  Getting_Started_Unity_Catalog.py     is_marimo_app = True
  NYC_Taxi_Pipeline.py                 is_marimo_app = True
  Verify_Spark_And_S3.py               is_marimo_app = True
  _nexus_spark.py                      is_marimo_app = False   ← helper module, correct
```

`marimo check` reports no new issues; the pre-existing `error[branch-expression]` in `Getting_Started_PySpark.py` is untouched and filed as #817.

## The guard

`tests/unit/test_workspace_seeds.py` replicates the rule against the **real** tree, not a fixture — what ships is the bytes on disk. It:

- refuses to pass vacuously if the seed directory is ever renamed away,
- skips helper modules that are not notebooks at all,
- asserts the two markers **separately**, so a failure names the one that fell out of the window rather than reporting "one of two things is wrong".

Checked by putting a long docstring back and watching it fail.

## Rules, so the next one is prevented rather than caught

The test catches the fault after it is written. `f048d92` adds what stops it being written: `CLAUDE.md` rule 7 for the agent, a `.coderabbit.yaml` path instruction and `copilot-instructions.md` section 10 for the PR reviewers. Each states the limit with its source, why it is worth a rule (nothing fails — the file just stops looking like a notebook), and the two things *not* to do: reach for `# /// script`, or flag a helper module.

Two dead references to `scripts/deploy.sh` in `copilot-instructions.md` are corrected while there — that file was removed in #505, and the reviewer was being pointed at it for who registers `system.flow-sync`. The orchestrator does (`src/nexus_deploy/kestra.py`).

## Local CodeRabbit round

Reviewed `b5be9b6`, **1 finding**, fixed:

- 🟠 *"any table you create is immediately query-able over HTTP"* in the PostgREST seed overstated it — PostgREST needs a schema-cache reload and the connecting role needs read access. The sentence is **verbatim from `main`**; this branch only moved it out of the docstring. The same notebook already explains the reload fifty lines further down, so the summary contradicted its own setup section. Qualified rather than deleted.

Not covered by that round: `f048d92`, which was written after it. The first attempt at the round does not count — it aborted with `WebSocket closed` before reviewing anything, and did so with exit code 0.

## How to test

Nothing to deploy for the rule changes. For the notebooks:

1. The fix reaches a stack whose workspace repo does **not** already hold the files — seeding is create-only (`POST`, 422 → skipped), so an existing `nexus_seeds/marimo/*.py` is preserved by design.
2. On a stack where they already exist, delete `nexus_seeds/marimo/*.py` in Forgejo and re-run `spin-up.yml`; or run `destroy-all` with **both** confirmation fields set to `DESTROY`, which recreates the fork from the GitHub mirror and re-seeds from scratch (verified: `mirror-seed-rerun: ok — created=14 skipped=0`).
3. Then check marimo's Workspace tree: all six notebooks should carry the notebook icon and be listed.

Delivering fixes to an *existing* seed file without a full destroy is a separate gap, deliberately not addressed here.

## Summary by Sourcery

Ensure seeded marimo notebooks appear as notebooks by keeping their recognition markers within the first 512 bytes and enforcing the convention in tests and contributor guidance.

Bug Fixes:
- Make all seeded marimo notebooks recognizable in marimo’s workspace and file browser by keeping their identifying markers within marimo’s header limit.

Enhancements:
- Move notebook introductions into the first markdown cell while retaining concise module summaries.
- Add unit coverage that validates real seeded notebooks against marimo’s 512-byte recognition rule and distinguishes helper modules.
- Document the header constraint and correct remediation across contributor and review guidance.

Documentation:
- Update contributor, Copilot, and CodeRabbit guidance with marimo notebook recognition requirements and correct the Kestra orchestrator reference.

Tests:
- Add tests covering recursive marimo seed discovery, non-empty seed protection, helper-module exclusion, and separate marker validation.

## Summary by CodeRabbit

- **Documentation**
  - Added guidance for maintaining marimo notebook compatibility and keeping required detection markers within the first 512 bytes.
  - Updated review instructions and developer documentation with notebook authoring recommendations.

- **Improvements**
  - Moved detailed notebook introductions and usage guidance into their opening markdown cells, preserving the information while improving notebook recognition.

- **Tests**
  - Added validation to ensure marimo notebooks are correctly detected and required markers remain within the supported header limit.

### CODE REVIEW

File: tests/unit/test_workspace_seeds.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import pytest
+
+SEEDS_DIR = Path(__file__).resolve().parents[2] / "examples" / "workspace-seeds"
+MARIMO_SEEDS = sorted((SEEDS_DIR / "marimo").glob("*.py"))
```
[the remark this answers is not in the document]
Comment (stefanko-ch) [reply tests/unit/test_workspace_seeds.py]: Fixed in b6e1a9f — valid, and the more consequential of the two.

The seeder is the tiebreaker: it walks `root.rglob("*")` (`src/nexus_deploy/seeder.py:170`), and other seed trees already nest — `kestra/flows`, `kestra/workflows`, `prefect/flows`. So a notebook at `marimo/<sub>/x.py` would have shipped **and** gone unchecked, while the `.coderabbit.yaml` path this same PR adds already says `marimo/**/*.py`. The test was the odd one out.

Now `rglob`. Verified by planting a nested notebook whose `marimo.App` lands at byte 897:

```
FAILED tests/unit/test_workspace_seeds.py::test_marimo_seed_is_recognisable_as_a_notebook[nested/Probe.py]
```

The parametrisation ids moved with it, from `p.name` to the path relative to the marimo seed dir — two nested files sharing a basename would otherwise collapse into one unreadable id. The failure above shows the new form.

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -118,10 +118,24 @@ What's not allowed:
...
 
 Convention background: [examples/README.md](../examples/README.md) carries the full rationale and the rules for the rest of the seed-tree (path-mapping, idempotency, secret references via `{{ secret('NAME') }}`, etc.).
 
+## 10. Seeded marimo notebooks — the 512-byte header
+
+marimo decides a `.py` file is a notebook by reading only the **first 512 bytes** and checking for both `import marimo` and `marimo.App` (`marimo/_server/files/directory_scanner.py`, `READ_LIMIT = 512`). A long module docstring pushes both markers past that window.
+
+**Flag any new or modified `.py` under `examples/workspace-seeds/marimo/` whose `marimo.App` lands after byte 512.**
```
[the remark this answers is not in the document]
Comment (stefanko-ch) [reply .github/copilot-instructions.md]: Fixed in b6e1a9f — valid, and there were two gaps in that line rather than one.

The paragraph above it states the rule correctly; the actionable instruction then narrowed it:

> Flag any new or modified `.py` … whose `marimo.App` lands after byte 512.

**One marker instead of two.** Sufficient in practice — a marimo-generated header always emits `import marimo` before `app = marimo.App(...)`, so the later marker is the binding one — but an instruction should state the condition, not a proxy that happens to hold.

**"lands after" is not the test.** The scanner reads `source[:512]` and asks whether each marker is *in* it, so a marker starting at byte 505 runs past the boundary and does not count. "Lands after byte 512" would pass it.

Both stated explicitly now, plus "including nested paths" to match the sibling fix on the test.
